File: ruby/person.rb
$HELLO_WORLD = "Hello, world"
FREDDY = "Freddy"

class Person

  @@height = []

  attr_accessor :name, :age

  def Person.add_height(height)
    unless @@height.include?(height)
      @@height << height
    end
  end

  def initialize(name, age)
    @name = name
    @age = age
    @height = "small"
  end

  def height=(height)
    unless @@height.include?(height)
      puts "Unknown height: #{height}."
    end
    @height = height
  end
end

class Italian < Person

  attr_reader :city

  def initialize(name, age, city="Venice")
    @city = city
    super(name, age)
  end

end

puts "Global #{$HELLO_WORLD}."

Person.add_height("small")
Person.add_height("medium")
Person.add_height("tall")

fred = Person.new("fred", 12)
fred.height = "medium"
puts fred.name
p fred

puts
geno = Italian.new("geno", 32)
puts geno.name
p geno

File: go/person.go
package main

import "fmt"

type Person struct {
	name string
	age  int
}

type Italian struct {
	Person
	region string
}

func changeAge(person *Person) {
	person.age += 10
}

func (person *Person) ageAdjust() {
	person.age += 20
}

func main() {
	fmt.Println("Hello, world.")

	var fred Person
	fred.name = "fred"
	fred.age = 22

	peter := Person{"peter", 34}

	fmt.Println("Before: ", fred)
	fmt.Println("Before: ", peter)

	changeAge(&fred)
	changeAge(&peter)

	fmt.Println("After: ", fred)
	fmt.Println("After: ", peter)

	fred.ageAdjust()
	peter.ageAdjust()

	fmt.Println("After method: ", fred)
	fmt.Println("After method:", peter)
}
